test/utils/index.ts

```diff
@@ -13,4 +15,309 @@ describe('utils', () => {
+
+		it('escape character with other character 4', () => {
+			// doesn't decode if illegal
+			const input = '<div>&copy1abc</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, input);
+		});
+
+		it('escape character with other character 5', () => {
+			const input = '<div>&copy other_chars</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>© other_chars</div>');
+		});
+
+		it('escape character with other character 6', () => {
+			const input = '<div>&copy; other_chars</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>© other_chars</div>');
+		});
+
+		it('escape character with other character 7', () => {
+			const input = '<div>other_chars&copy; other_chars</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>other_chars© other_chars</div>');
+		});
+
+		it('escape character with other character 8', () => {
+			const input = '<div>other_chars&copy; other_chars&copy</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>other_chars© other_chars©</div>');
+		});
+
+		it('escape character with other character 9', () => {
+			const input = '<div>&copy=other_chars</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>©=other_chars</div>');
+		});
+
+		it('escape character with other character 10', () => {
+			const input = '<div>&rect;</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, '<div>▭</div>');
+		});
+
+		it('escape character with other character 11', () => {
+			// doesn't decode '&rect' which isn't html entity
+			const input = '<div>&rect</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, input);
+		});
+
+		it('escape character with other character 12', () => {
+			// doesn't decode '&DownArrowUpArrow' which isn't html entity
+			const input = '<div>&DownArrowUpArrow</div>';
+			const value = decode_character_references(input);
+			assert.equal(value, input);
+		});
+
+		it('validate code 1', () => {
+			// line feed becomes generic whitespace
+			const input = '&#10;';
+			const value = decode_character_references(input);
+			assert.equal(value, ' ');
+		});
+
+		it('validate code 2', () => {
+			// code points 128-159
+			const input = '&#159;';
+			const value = decode_character_references(input);
+			// 376 from windows_1252 of src/compiler/parse/utils/html.ts
+			assert.equal(value, String.fromCodePoint(376));
+		});
+
+		it('validate code 3', () => {
+			// UTF-16 surrogate halves 55297 - 57343
+			const input = '&#57343;';
+			const value = decode_character_references(input);
+			assert.equal(value, String.fromCodePoint(0));
+		});
+
+		it('validate code 4', () => {
+			// out of max range
+			const input = '&#196608;';
+			const value = decode_character_references(input);
+			assert.equal(value, String.fromCodePoint(0));
+		});
```

Comment:
If it renders with plain HTML, the result is different between HTML and Svelte.
Do we need to adjust them?

https://codepen.io/baseballyama/pen/bGKQwvz

Reply:
I'm a bit unclear about the purpose of this processing.But it is old logic, i just fill test to it.I think it's best not to modify this logic.If there is a related issue by then, fix it.